src/serialize/json/normalize.ts

```diff
@@ -446,7 +447,7 @@ export class LicenseNormalizer extends BaseJsonNormalizer<Models.License> {
           ? undefined
           : this._factory.makeForAttachment().normalize(data.text, options),
         url: JsonSchema.isIriReference(url)
-          ? url
+          ? percentEncodeUrl(url)
```

Comment:
the encoding should have happened before the `JsonSchema.isIriReference()` is run on it.